include/seastar/core/metrics_api.hh

```diff
@@ -323,7 +324,7 @@ public:
 
 using value_map = std::map<sstring, metric_family>;
 
-using metric_metadata_vector = std::vector<metric_info>;
+using metric_metadata_fifo = chunked_fifo<metric_info>;
```

Comment:
Isn't std::deque more suitable here?

Reply:
I argue for `chunked_fifo` because `std::deque` still does some contiguous allocations since it uses chunks of arrays. The problem that we were trying to address here was an overallocation issue on contiguous allocations.

May I ask, what is your concern? @avikivity

Reply:
> I argue for `chunked_fifo` because `std::deque` still does some contiguous allocations since it uses chunks of arrays. The problem that we were trying to address here was an overallocation issue on contiguous allocations.
> 
> May I ask, what is your concern? @avikivity

After posting I realized that `std::deque` is a good substitute for `values_copy::values` because it needs random access at locations [scollectd.cc:429](https://github.com/scylladb/seastar/blob/28daa8f50ccd5737b1779a2656d4330473d53d59/src/core/scollectd.cc#L429), [scollectd.cc:450](https://github.com/scylladb/seastar/blob/28daa8f50ccd5737b1779a2656d4330473d53d59/src/core/scollectd.cc#L450), [scollectd.cc:471](https://github.com/scylladb/seastar/blob/28daa8f50ccd5737b1779a2656d4330473d53d59/src/core/scollectd.cc#L471) and `chunked_fifo` does not support random access.

Reply:
Mostly that std::queue is a standard type and more similar to std::vector.

Reply:
> Mostly that std::queue is a standard type and more similar to std::vector.

Thank you for the response. I am a little confused though, are there more items to discuss? cc: @avikivity

Reply:
> Mostly that std::queue is a standard type and more similar to std::vector.

These types are similar and they do have a design/interface that is unique compared to `chunked_fifo`. However, the variables `_current_metrics` and `metric_metadata_vector` do not seem to rely on those differences. A `chucked_fifo` also decreases the potential of memory fragmentation. For these reasons, I believe a `chunked_fifo` is OK.

I also believe a `std::deque` is OK for those places that could use improved memory management but still need the interfaces from `std`,  (e.g., `values_copy::values`).

Reply:
I think deque is more suitable, chunked_fifo is a specialized type while deque is general-purpose.

Reply:
> I think deque is more suitable, chunked_fifo is a specialized type while deque is general-purpose.

@avikivity I am a little confused, are you requesting changes here?

Reply:
Yes, please switch to deque.

Reply:
Made the change.